### PR TITLE
make index record mutable again

### PR DESCRIPTION
**Pull request by kalefranz**

### CODE REVIEW

File: conda/core/index.py [hunk 1]
```diff
@@ -493,13 +497,6 @@ def add_http_value_to_dict(resp, http_key, d, dict_key):
         d[dict_key] = value
 
 
-def add_pip_dependency(index):
-    # TODO: discuss with @mcg1969 and document
-    for dist, info in iteritems(index):
```
Comment (kalefranz) [inline conda/core/index.py]: Also note here, taking out the "table scan" just to satisfy the add_pip stuff

**Comment by kalefranz** [comment]
compare

https://travis-ci.org/conda/conda/jobs/196291361#L636

to

https://travis-ci.org/conda/conda/jobs/196284275#L636

70 seconds vs 147 seconds

Hopefully this will take the edge off.  @mcg1969 if you have anything useful in https://github.com/conda/conda/pull/4456, we should also include that

**Comment by kalefranz** [comment]
Now I'm confused again.  Comparing to the 4.3.7 tag total build times, this didn't really help at all.
https://travis-ci.org/conda/conda/builds/193861264

**Comment by kalefranz** [comment]
Ok, so it's not fair to compare conda 4.3 build time to conda 4.2.  Conda 4.3 has a lot of extra tests and coverage, including almost complete integration tests for `conda clean`, which dramatically increases the total test time.

I've created https://github.com/conda/conda/pull/4462, which we can use to compare how both conda 4.2.x and conda 4.3.x compare when tested against conda-build 2.1.2.  Also not 100% equivalent because of some ignored tests for conda 4.3, but it's closer I think.

**Comment by mcg1969** [comment]
I'd like to see this merged. Then I'd like to build some of my #4456 work on top of this. I think there are about three PR's worth of changes in #4456 that should be submitted and considered separately.

**Pull request merged**